File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> preProduct(nums.size(),0), postProduct(nums.size(),0);
        preProduct[0] = nums[0];
        for(int i = 1; i < nums.size(); ++i){
            preProduct[i] = preProduct[i - 1] * nums[i];
        }
        postProduct[nums.size()-1] = nums[nums.size()-1];
        for(int i = nums.size()-2; i>=0;--i){
            postProduct[i] = postProduct[i+1] * nums[i];
        }

        vector<int> ans(nums.size(),0);
        for(int i = 0; i < nums.size(); ++i){
            if(i == 0){
                ans[i] = postProduct[i + 1];
            }else if(i == nums.size() - 1){
                ans[i] = preProduct[i - 1];
            }else{
                ans[i] = preProduct[i-1] * postProduct[i+1];
            }
        }
        return ans;

    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def recur(self, nums: List[int], target: int, index: int, current: List[int], ans: List[List[int]]):
        if target == 0:
            ans.append(list(current))
            return
        if index == len(nums):
            return
        if target < 0:
            return
        current.append(nums[index])
        self.recur(nums, target-nums[index], index, current, ans)
        #self.recur(nums, target-nums[index], index + 1, current, ans)
        current.pop()
        self.recur(nums, target, index + 1, current, ans)
        
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        ans = []
        current = []
        self.recur(nums, target, 0, current, ans)
        return ans